implementation/src/main/java/io/smallrye/health/checks/InetAddressHealthCheck.java

```diff
@@ -0,0 +1,69 @@
+package io.smallrye.health.checks;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+
+/**
+ * Health check implementation to check if host is reachable using {@code java.net.InetAddress.isReachable} method.
+ *
+ * <pre>
+ * {@code
+ * @Produces
+ * @ApplicationScoped
+ * @Liveness
+ * HealthCheck check1() {
+ *   return new InetAddressHealthCheck("service.com");
+ * }
+ * }
+ * </pre>
+ *
+ * @see InetAddress
+ */
+public class InetAddressHealthCheck implements HealthCheck {
+
+    private static final String DEFAULT_NAME = "Internet Address Check";
+    private static final int DEFAULT_TIMEOUT = 2000;
+
+    private String host;
+    private String name;
+    private int timeout;
+
+    public InetAddressHealthCheck(String host) {
+        this.host = host;
+        this.name = DEFAULT_NAME;
+        this.timeout = DEFAULT_TIMEOUT;
+    }
+
+    @Override
+    public HealthCheckResponse call() {
+        final HealthCheckResponseBuilder healthCheckResponseBuilder = HealthCheckResponse.named(this.name);
+        healthCheckResponseBuilder.withData("host", this.host);
+        try {
+            InetAddress addr = InetAddress.getByName(this.host);
+            final boolean reachable = addr.isReachable(this.timeout);
+            if (!reachable) {
+                healthCheckResponseBuilder.withData("error", String.format("Host %s not reachable.", this.host));
+            }
+
+            healthCheckResponseBuilder.state(reachable);
+        } catch (IOException e) {
+            healthCheckResponseBuilder.withData("error", e.getMessage());
+            healthCheckResponseBuilder.down();
+        }
+
+        return healthCheckResponseBuilder.build();
+    }
+
```

Comment:
Could you add a small javadoc for `timeout` that indicates the unit the timeout needs to be specified in? Millis, etc